### ISSUE TITLE
Extra walls being generated by modifier

### ISSUE THREAD

**Issue by foresterr** [SoftFever/OrcaSlicer#5185]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### OrcaSlicer Version

2.0.0

### Operating System (OS)

Windows

### OS Version

Windows 10

### Printer

Creality Ender 3 v3 KE

### How to reproduce

Create a very simple model - for example a cube with a vertical hole through it (can be done entirely by use of primitives without leaving Orca). Add a modifier which has the entire hole inside it, but does not touch outer cube walls. Change any modifier setting related to walls (for example number of Wall Loops). Slice the model.

### Actual results

Additional walls are added at the modifier boundary.

### Expected results

Walls inside the modifier are altered according to the modifier (e.g. they have a different number of wall loops), but no extra walls are generated in the sliced model where the modifier ends. Maybe this is working as intended to avoid broken wall loops if the modifier boundary intersects a wall loop, but in case of wall loops fully enclosed inside a modifier, the result is rather confusing.

### Project file & Debug log uploads

[modifier potential bug.zip](https://github.com/SoftFever/OrcaSlicer/files/15142574/modifier.potential.bug.zip)

### Checklist of files to include

- [ ] Log file
- [X] Project file

### Anything else?

![2024-04-28 17_15_59-Window](https://github.com/SoftFever/OrcaSlicer/assets/6168870/df686b3c-fcef-478a-85ad-d7ca2cfe745e)

There is another bug issue (https://github.com/SoftFever/OrcaSlicer/issues/3097) that might be about the same thing, but it's closed as stale and tbh I can't figure out if the reporter there talks about the same thing (it's a bit confusingly written and illustrated).

**Comment by FDFD1313** [comment]
Same here, the modifier adds walls, instead of only changing the region walls, (is changing the region and, additionally, introducing new undesired walls)
![image](https://github.com/SoftFever/OrcaSlicer/assets/23555343/afc8e0cf-7640-47d5-9b5e-cc84770bf6ee)

**Comment by rocketscott** [comment]
bump

**Comment by riloum** [comment]
same here

**Comment by ProfessorKaos** [comment]
PIng to this

**Comment by TylonHH** [comment]
yeah that realy sucks
![grafik](https://github.com/user-attachments/assets/fa4a16b4-69dd-4f3a-bb20-008d7e83d6d5)

**Comment by Misterke** [comment]
Bump.  Note that this problem also exists in upstream PrusaSlicer.  That doesn't make it any less annoying though.  It's almost impossible now to use fuzzy skin but have horizonal screw holes excluded from the fuzziness.  Having a modifier over those holes adds annoying walls with the most outer ones still having fuzziness applied.

**Comment by Peteeeeeeeeee** [comment]
also exists in Bambu Studio and is driving me insane.

**Comment by irowboat** [comment]
I think this is what I’m seeing with my multi-color print right now. As a simple test, I’ve painted some spots on the outside surface of the object, and the colors run into the layer as I’d expect. What I **don’t** expect is for each of the _**internal boundaries**_ between colors to have extra perimeters or fuzzy skin. It seems like color is just another modifier as far as Orca is concerned, and is generating these internal walls—_with all the parameters_—drastically slowing the print (doing fuzzy skin on internal perimeters especially).

This has 4 perimeters (more perimeters = more fun) and a really fuzzy skin (0.02mm/1mm). It isn't dependent on classic or Arachne from what I can tell, but WOW does it affect the new beam interlocking if you've also got fuzzy skin:

<img width="614" alt="OrcaSlicer - issue #5185 - without_beam_interlocking" src="https://github.com/user-attachments/assets/a8551381-b610-42c8-9f74-23cf78f78030">

without beam interlocking

<img width="614" alt="OrcaSlicer - issue #5185 - with_beam_interlocking" src="https://github.com/user-attachments/assets/b13b1dfd-47f2-4339-bc20-981eb5e010bb">

**_with_** beam interlocking


#### Example file
[OrcaSlicer - issue #5185.3mf.zip](https://github.com/user-attachments/files/17047150/OrcaSlicer.-.issue.5185.3mf.zip)

**Comment by Ishprecher** [comment]
This is a problem that dates back to Slic3r apparently. Probably won't be fixed on Orca or Bambu until Prusaslicer does, but isn't likely since they've known about it for years now.
https://github.com/prusa3d/PrusaSlicer/issues/8121

**Comment by BSKFeralLycan** [comment]
Yeah it's still happening it's happening on Orca ports too It also happens on purpose laser I don't know why they haven't fixed this yet.. I'm also confused on why they've already closed your issue here?

Have you seen my photo there set for screw holes in three they're unmodified and the one that I'm trying to reinforce has created its own section this is also extremely annoying when you're just trying to do something like fuzzy skin because even though you're just trying to deal with the outside walls it's still creates a perimeter and mind you I have zero perimeters chosen so this should be zero. 

Off topic but I really wish that they would just make a paint tool for fuzzy skin and modifiers that only affect the outer perimeters

![Image](https://github.com/user-attachments/assets/a0b1b8e6-cedd-40e9-a943-ae172d0c4345)

**Comment by Misterke** [comment]
Latest Orca at least fixed the problem when having a modifier just applying fuzzy skin and also when just modifying the amount or type of infill, but if you change the number of walls (and probably some other parameters affecting walls) then indeed additional walls are added at the boundary of the modifier.  I myself am already very happy that for infill and  fuzzy skin I can now just create modifier objects in CAD without then getting extra walls weakening those parts ...

**Comment by Waldehyd** [comment]
Just ran in to this on 2.3.0
Weird how long this bug seems to exist.

**Comment by m-sundman** [comment]
> if you add a parameter to the modifier which does add those extra wall, slice and then remove that parameter again (so restore it to default), then still extra wall are generated when you slice - even though the settings are the same as what before was not giving you extra walls. If you then disable the entire Assembly and then reenable it and reslice, then you get back to not having those extra walls

This indicates that this problem was fixed in the wrong way, fixing some symptoms instead of fixing the cause. Good catch!

And @Waldehyd yes, it's super annoying that bugs are getting closed just because no dev bothered to fix it. At the very least they should get automatically reopened when comments are added.

**Comment by LIFESJESTER** [comment]
I was just having the same problem and found this post while looking for a answers. After about 20 changes and saving and reopening to no avail, I finally found a "solution"... 

Press UNDO to the last place it didn't add the perimeters. 

Only had to hit undo about 50 times before I finally got there due to what I will call troubleshooting. I wanted to see if it might be a viable workaround so I selected REDO and it went back to adding the perimeters. Pressed UNDO once again and they were gone. Hopefully it was not just a fluke and next time will only have to use UNDO once as soon as the issue appears and it works again.

***Side note, I noticed that if the Wall Loops number on the modifiier was different from the main setting, it created the issue.

**Comment by YashMathur** [comment]
@LIFESJESTER +1 to that - it only happens to me when wall loops are modified from the main setting. 

I haven't found a solution for this issue but have a workaround. I split my main body into 2 parts in fusion (can be done in orca if its simple enough) and used height range modifiers to modify the wall loops where needed.

**Comment by Kemerd** [comment]
Getting this issue too!

**Comment by MrMonkey1302** [comment]
Is there someone looking at fixing this. It's still an issue as of 2.4.0 Alpha
